scarb/src/compiler/plugin/proc_macro/host.rs

```diff
@@ -1,34 +1,45 @@
 use crate::compiler::plugin::proc_macro::{Expansion, FromSyntaxNode, ProcMacroInstance};
 use crate::core::{Config, Package, PackageId};
 use anyhow::{ensure, Result};
-use cairo_lang_defs::db::DefsGroup;
+use cairo_lang_defs::ids::{ModuleItemId, TopLevelLanguageElementId};
 use cairo_lang_defs::plugin::{
     DynGeneratedFileAuxData, GeneratedFileAuxData, MacroPlugin, MacroPluginMetadata,
     PluginGeneratedFile, PluginResult,
 };
 use cairo_lang_defs::plugin::{InlineMacroExprPlugin, InlinePluginResult, PluginDiagnostic};
+use cairo_lang_diagnostics::ToOption;
 use cairo_lang_macro::{
-    AuxData, Diagnostic, ExpansionKind, ProcMacroResult, Severity, TokenStream, TokenStreamMetadata,
+    AuxData, Diagnostic, ExpansionKind, FullPathMarker, ProcMacroResult, Severity, TokenStream,
+    TokenStreamMetadata,
 };
+use cairo_lang_semantic::db::SemanticGroup;
+use cairo_lang_semantic::items::attribute::SemanticQueryAttrs;
 use cairo_lang_semantic::plugin::PluginSuite;
-use cairo_lang_syntax::attribute::structured::AttributeListStructurize;
+use cairo_lang_syntax::attribute::structured::{
+    Attribute, AttributeArgVariant, AttributeListStructurize,
+};
+use cairo_lang_syntax::node::ast::Expr;
 use cairo_lang_syntax::node::db::SyntaxGroup;
 use cairo_lang_syntax::node::ids::SyntaxStablePtrId;
 use cairo_lang_syntax::node::{ast, TypedSyntaxNode};
 use itertools::Itertools;
 use scarb_stable_hash::short_hash;
 use std::any::Any;
-use std::sync::Arc;
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
 use std::vec::IntoIter;
 use tracing::{debug, trace_span};
 
+const FULL_PATH_MARKER_KEY: &str = "full_path_marker";
```

Comment:
What do you think about namespacing macro-specific attrs, like `starknet` does?

```suggestion
const FULL_PATH_MARKER_KEY: &str = "macro::full_path_marker";
```